examples/chip-tool/commands/tests/TestCommand.cpp

```diff
@@ -20,46 +20,23 @@
 
 constexpr uint16_t kWaitDurationInSeconds = 30;
 
-static void test_os_sleep_ms(uint64_t millisecs)
+CHIP_ERROR TestCommand::Run(NodeId localId, NodeId remoteId)
 {
-    struct timespec sleep_time;
-    uint64_t s = millisecs / 1000;
+    CHIP_ERROR err = CHIP_NO_ERROR;
 
-    millisecs -= s * 1000;
-    sleep_time.tv_sec  = static_cast<time_t>(s);
-    sleep_time.tv_nsec = static_cast<long>(millisecs * 1000000);
+    chip::DeviceLayer::PlatformMgr().LockChipStack();
+    err = GetExecContext()->Commissioner->GetDevice(remoteId, &mDevice);
+    chip::DeviceLayer::PlatformMgr().UnlockChipStack();
+    ReturnErrorOnFailure(err);
 
-    nanosleep(&sleep_time, nullptr);
-}
-
-CHIP_ERROR TestCommand::Run(PersistentStorage & storage, NodeId localId, NodeId remoteId)
-{
-    ReturnErrorOnFailure(mOpCredsIssuer.Initialize(storage));
-
-    chip::Controller::CommissionerInitParams params;
-
-    params.storageDelegate                = &storage;
-    params.operationalCredentialsDelegate = &mOpCredsIssuer;
-
-    ReturnErrorOnFailure(mCommissioner.SetUdpListenPort(storage.GetListenPort()));
-    ReturnErrorOnFailure(mCommissioner.Init(localId, params));
-    ReturnErrorOnFailure(mCommissioner.ServiceEvents());
-    ReturnErrorOnFailure(mCommissioner.GetDevice(remoteId, &mDevice));
-
-    ReturnErrorOnFailure(NextTest());
+    chip::DeviceLayer::PlatformMgr().LockChipStack();
+    err = NextTest();
```

Comment:
Curious why the locking is around here. It's not a stack call or obviously concurrent access, so the locking should maybe be moved down into NextTest

Reply:
Yeah, will move it one layer down into the actual implementation of NextTest (which in turn, contains various tests that send out commands which need protection).

Reply:
I actually cannot move it a layer down on second glance, since NextTest() is called by the response callbacks as well to run the next test automatically. These callbacks are being dispatched from the chip thread, which have the lock held.